src/libbson/src/bson/bson-json.c

```diff
@@ -735,6 +741,15 @@ _bson_json_parse_double (bson_json_reader_t *reader,
    *d = strtod (val, NULL);
 
 #ifdef _MSC_VER
+#ifdef INFINITY
+   const double pos_inf = INFINITY;
+   const double neg_inf = -pos_inf;
+#else
+   const unsigned long inf_rep[2] = {0x00000000, 0x7ff00000};
+   const double pos_inf = *(double *) inf;
```

Comment:
I believe you are correct, but this was left unchanged to minimize changes to the warnings being addressed. I can apply a drive-by improvement to avoid the strict aliasing issue if preferrable.

Reply:
Removed the code entirely as check for `INFINITY` (and `NAN`) is no longer applicable (we do not support VS 2010 or older).